### PR TITLE
Fix: Remove the invalid license key notice for Products with empty keys

### PR DESCRIPTION
**Pull request by nefeline**
Ref. https://central.tri.be/issues/115562

### CODE REVIEW

File: src/Tribe/PUE/Notices.php [hunk 1]
```diff
@@ -197,6 +197,49 @@ public function setup_notices() {
...
+		$sql = "
+			SELECT option_name
+				FROM {$wpdb->options}
+				WHERE option_name LIKE 'pue_install_key_%'
+					AND option_value=''
+			";
+
+		$empty_keys = $wpdb->get_results( $sql, ARRAY_N );
+
+		$plugin_names = [
```
Comment (juanfra) [inline src/Tribe/PUE/Notices.php]: Please use `array()` so we can be compatible with the PHP versions we need to be.

File: src/Tribe/PUE/Notices.php [hunk 2]
```diff
@@ -208,6 +251,17 @@ public function setup_notices() {
 	public function render_invalid_key() {
 		global $pagenow;
 
+		$empty_keys = $this->select_empty_keys();
+
+		// Remove the invalid_key notice for products with an empty license key
+		if ( ! empty( $empty_keys ) ) {
```
Comment (juanfra) [inline src/Tribe/PUE/Notices.php]: Invert the logic of the `empty` and return to avoid indentation

File: src/Tribe/PUE/Notices.php [hunk 3]
```diff
@@ -197,6 +197,49 @@ public function setup_notices() {
...
+			'pue_install_key_events_community'         => 'The Events Calendar: Community Events',
+			'pue_install_key_events_community_tickets' => 'The Events Calendar: Community Events Tickets',
+			'pue_install_key_image_widget_plus'        => 'Image Widget Plus',
+			'pue_install_key_tribe_eventbrite'         => 'The Events Calendar: Eventbrite Tickets',
+			'pue_install_key_tribe_filterbar'          => 'The Events Calendar: Filter Bar',
+			'pue_install_key_event_aggregator'         => 'Event Aggregator',
+			'pue_install_key_events_calendar_pro'      => 'The Events Calendar PRO',
+		];
+
+		$formatted_empty_keys = [];
```
Comment (juanfra) [inline src/Tribe/PUE/Notices.php]: Use `array()` here as well.

File: src/Tribe/PUE/Notices.php [hunk 4]
```diff
@@ -197,6 +197,49 @@ public function setup_notices() {
...
+		foreach ( $empty_keys as $empty_key ) {
+			if ( isset( $plugin_names[ $empty_key[0] ] ) ) {
```
Comment (juanfra) [inline src/Tribe/PUE/Notices.php]: We can also use the array utils here.

File: src/Tribe/PUE/Notices.php (hunk 1 above)
Comment (nefeline) [reply src/Tribe/PUE/Notices.php]: :+1:

File: src/Tribe/PUE/Notices.php (hunk 2 above)
Comment (nefeline) [reply src/Tribe/PUE/Notices.php]: :zap: Done!

File: src/Tribe/PUE/Notices.php (hunk 3 above)
Comment (nefeline) [reply src/Tribe/PUE/Notices.php]: :+1:

**Comment by juanfra** [comment]
Please be sure to resolve the readme conflicts before merge

**Pull request merged**